signing/secp256k1/impl/src/main/java/tech/pegasys/signers/secp256k1/multikey/MultiKeySignerProvider.java

```diff
@@ -73,9 +74,21 @@ public MultiKeySignerProvider(
 
   @Override
   public Optional<Signer> getSigner(final PublicKey publicKey) {
-    return signingMetadataTomlConfigLoader
-        .loadMetadata(configFileSelector.getSpecificConfigFileFilter(publicKey))
-        .map(metadataFile -> metadataFile.createSigner(this));
+    final Optional<Signer> signer =
+        signingMetadataTomlConfigLoader
+            .loadMetadata(configFileSelector.getSpecificConfigFileFilter(publicKey))
+            .map(metadataFile -> metadataFile.createSigner(this));
+    if (signer.isPresent()) {
+      if (Arrays.equals(signer.get().getPublicKey().getValue(), publicKey.getValue())) {
```

Comment:
This shouldn't be necessary to use Arrays.equals as the PublicKey type implements equals

Reply:
The _base_ publicKey type doesn't offer equals - only the PublickKeyImpl - which is not _necessarily_ being used - thus you need to use the interface being offered by the PublicKey type, no?

Reply:
Since there a interface yes, we can't guarantee that is implemented